integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -254,6 +254,16 @@ def post(self, request, *args, **kwargs):
                 page_translation_form.instance.page = page_form.save()
             # Save page translation form
             page_translation_form.save()
+            # If any source translation changes to draft, set all depending translations/versions to draft
+            if page_translation_form.instance.status == status.DRAFT:
+                language_tree_node = region.language_node_by_id.get(language.id)
+                if not language_tree_node.is_leaf():
+                    partial_language_tree = LanguageTreeNode.get_tree(
+                        parent=language_tree_node
+                    )
+                    page_translation_form.instance.page.translations.filter(
+                        language__in=[node.language for node in partial_language_tree]
+                    ).update(status=status.DRAFT)
```

Comment:
```suggestion
                languages = [language] + [
                    node.language for node in language_tree_node.get_descendants()
                ]
                page_translation_form.instance.page.translations.filter(
                    language__in=languages
                ).update(status=status.DRAFT)
```
Oh, I'm sorry, my suggestion had a flaw - in case the language tree node is a leaf, we need to update all its translations anyway, even if there are no dependent other languages.
Also, using `get_tree()` instead of `get_descendants()` uses one additional query for leafs (and since we did query the current language from the database earlier, it makes no sense to do it again)... so I'd go back to your initial proposal of using `get_descendants()` :see_no_evil: